# Recommender Systems - Data Mining Lab 2

In [2]:
import scipy
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import sys

### Data loading

In [3]:
train_file_path = 'data/lab2_train.csv' 
test_file_path = 'data/lab2_test.csv'

train_data = pd.read_csv(train_file_path, delimiter=',')
test_data = pd.read_csv(test_file_path, delimiter=',')


In [26]:
#print(train_data)
like_train_data = train_data[train_data.is_like == True]
like_train_data = like_train_data.sort_values("user_to_id")
liked_user_count = like_train_data["user_to_id"].nunique()
user_count = like_train_data["user_from_id"].nunique()
print(user_count)
print(liked_user_count)
print(like_train_data)
match_train_data = train_data[train_data.is_match ==True]
match_train_data = match_train_data.sort_values("user_from_id")
print(match_train_data)

2107
1707
       user_from_id  user_to_id  is_like is_match
50598            93           2     True    False
46348            24           2     True    False
11643            88           2     True    False
61229            82           2     True    False
71280            67           2     True    False
...             ...         ...      ...      ...
32426          1220        3623     True    False
55939          1022        3624     True    False
5109            322        3624     True    False
14716           236        3624     True      NaN
23406          1022        3624     True    False

[12637 rows x 4 columns]
       user_from_id  user_to_id  is_like is_match
62638             1        3046     True     True
40079             4        3058     True     True
676              19        2550     True     True
73991            36        3335     True     True
8940             44         150     True     True
...             ...         ...      ...      ...
8137          

## Familiarization
### Important properties of the data

### Types of people
There are only 1707 user who have one like or more of the 2107 users who gave likes. Of all of these users there are only 415 matches.